osfoffline/polling_osf_manager/polling.py

```diff
@@ -29,42 +31,84 @@ def __init__(self, user, loop):
         self.user = user
 
         self._loop = loop
+        self.poll_job = None
+        self.process_job = None
         self.osf_query = OSFQuery(loop=self._loop, oauth_token=self.user.oauth_token)
-        self.polling_event_queue = PollingEventQueue(loop=self._loop)
 
     def stop(self):
+        logger.info('OSF polling requested to stop.')
+
+        # Stop was called before start could complete
+        if not self.poll_job or not self.process_job:
+            return
+
+        # If these futures are complete an exception has been raised in them
+        if self.poll_job.done():
+            raise self.poll_job.exception()
+        if self.process_job.done():
+            raise self.process_job.exception()
 
-        self._keep_running = False
+        # Other wise cancel them, the event loop will clean up for us
+        self.poll_job.cancel()
+        self.process_job.cancel()
         self.osf_query.close()
+        logger.info('OSF polling requested stopped.')
+
+    def handle_exception(self, future):
```

Comment:
I believe that futures are the future. (Let's be honest here; the children had their chance) 

This does appear to simplify a lot of the request cleanup; we'll regression test this heavily of course.